### ISSUE TITLE
Add YUL support for top level functions

### ISSUE THREAD

**Issue by montyly** [crytic/slither#899]
```solidity
// function from https://docs.soliditylang.org/en/v0.8.0/assembly.html
function at(address _addr) view returns (bytes memory o_code) {
    assembly {
        // retrieve the size of the code, this needs assembly
        let size := extcodesize(_addr)
        // allocate output byte array - this could also be done without assembly
        // by using o_code = new bytes(size)
        o_code := mload(0x40)
        // new "memory end" including padding
        mstore(0x40, add(o_code, and(add(add(size, 0x20), 0x1f), not(0x1f))))
        // store length in memory
        mstore(o_code, size)
        // actually retrieve the code, this needs assembly
        extcodecopy(_addr, add(o_code, 0x20), 0, size)
    }
}


contract A{}
```

`YulScope` assumes there is always a contract (https://github.com/crytic/slither/blob/master/slither/solc_parsing/yul/parse_yul.py#L140), but this is not true for top-level function

**Comment by montyly** [comment]
Hi @hhamud. We took a bit of delay on this, and we won't be able to work make progresses for the next two weeks. But I am hopping to get this fixed in December

**Comment by montyly** [comment]
Fixed with #987
